victoriensularieh.ConsoleHabitTracker/Menu.cs

```diff
@@ -0,0 +1,464 @@
+            switch (choice.ToUpper())
+            {
+                case "A":
+                    ShowHabitMenu();
+                    rightChoice = true;
+                    break;
+                case "B":
+                    ShowUnitMenu();
+                    rightChoice = true;
+                    break;
+                case "C":
+                    ShowEntryMenu();
+                    rightChoice = true;
+                    break;
+                case "X":
+                    rightChoice = true;
+                    break;
+                default:
+                    Console.WriteLine(@"Wrong choice. Try again please.");
+                    break;
+            }
+        }
+    }
+
+    public static void ShowHabitMenu()
+    {
+        string choice = "";
+        bool rightChoice = false;
+        int habitId;
+        while (rightChoice == false)
+        {
+            Console.Clear();
+            Console.WriteLine("Welcome to the Habits Menu.");
+            Console.WriteLine("---------------------------");
+            Console.WriteLine(@"Press 1 To View Tracked Habits
+Press 2 to add a habit.
+Press 3 to update a habit.
+Press 4 to delete a habit.
+Press X to go back to the main menu.");
+            choice = Helpers.ValidateStringInput();
+            switch (choice.ToUpper())
+            {
+                case "1":
+                    Console.Clear();
+                    Habit.DisplayHabits();
+                    Console.WriteLine("Press any key to go back.");
+                    Console.ReadLine();
+                    break;
+                case "2":
+                    Console.Clear();
+                    Console.WriteLine("Enter the name of your Habit:");
+                    string habitName = Helpers.ValidateStringInput();
+                    Console.WriteLine("Choose the ID of the unit to use:");
+                    Unit.DisplayUnits();
+                    int habitUnit = Helpers.ValidateIntegerInput();
+                    while (!Unit.UnitExist(habitUnit))
+                    {
+                        Console.WriteLine("No Unit with the chosen ID. Choose again.");
+                        habitUnit = Convert.ToInt32(Console.ReadLine());
+                    }
+                    if (Habit.AddHabit(habitName, habitUnit))
+                    {
+                        Console.ForegroundColor = ConsoleColor.Green;
+                        Console.WriteLine("Habit inserted successfully.");
+                        Console.ResetColor();
+                    }
+                    else
+                    {
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        Console.WriteLine("Something went wrong while inserting new Habit.");
+                        Console.ResetColor();
+                    }
+                    Console.WriteLine("Press any key to go back.");
+                    Console.ReadLine();
+                    break;
```

Comment:
i tried to call the models methods (regardless if they do the DB access or not) inside the Menu (considered my frontend) to simulate an API call, when the user choose the model to deal with (or click a button) I call some sort of API that return a success or failure message(true or false in this application)

Reply:
I see what you mean with regards to using the Model classes as endpoints of sorts, but I would maintain that this amount of processes should be carried out outside of the switch statement cases. It doesn't necessarily need to be in a separate class, although if this class is going to be called `Menu` then I would question whether CRUD operations are appropriate in a Menu class.

Reply:
if I understand you it would go like this:
Menu class would have a method to display the text, another method to handle the choice of the user (which will include the switch statement), and then inside the choice method I will call the delete code (another method inside the model class) which in turn will call a DB class that handle the queries
Menu.DisplayMenu() -> Menu.UserChoice() -> Habit.AddHabit -> Database.InsertHabit()
something like this flow